## 38_kbsd.patch by Petr Salinger <Petr.Salinger@seznam.cz>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: GNU/kFreeBSD support. bz#356011
## DP: Added Hurd support.

Index: nss/nss/lib/freebl/unix_rand.c
===================================================================
--- nss.orig/nss/lib/freebl/unix_rand.c	2014-01-13 07:31:45.939967858 +0900
+++ nss/nss/lib/freebl/unix_rand.c	2014-01-13 07:31:45.935967956 +0900
@@ -156,7 +156,8 @@
 
 #if defined(SCO) || defined(UNIXWARE) || defined(BSDI) || defined(FREEBSD) \
     || defined(NETBSD) || defined(DARWIN) || defined(OPENBSD) \
-    || defined(NTO) || defined(__riscos__)
+    || defined(NTO) || defined(__riscos__) || defined(__GNU__) \
+    || defined(__FreeBSD_kernel__) || defined(__NetBSD_kernel__)
 #include <sys/times.h>
 
 #define getdtablesize() sysconf(_SC_OPEN_MAX)
Index: nss/nss/lib/softoken/softoken.h
===================================================================
--- nss.orig/nss/lib/softoken/softoken.h	2014-01-13 07:31:45.939967858 +0900
+++ nss/nss/lib/softoken/softoken.h	2014-01-13 07:31:45.935967956 +0900
@@ -197,7 +197,7 @@
 
 #define CHECK_FORK_MIXED
 
-#elif defined(LINUX)
+#elif defined(LINUX) || defined (__GLIBC__)
 
 #define CHECK_FORK_PTHREAD
 
Index: nss/nss/lib/ssl/sslmutex.c
===================================================================
--- nss.orig/nss/lib/ssl/sslmutex.c	2014-01-13 07:31:45.939967858 +0900
+++ nss/nss/lib/ssl/sslmutex.c	2014-01-13 07:31:45.935967956 +0900
@@ -56,7 +56,7 @@
     return SECSuccess;
 }
 
-#if defined(LINUX) || defined(AIX) || defined(BEOS) || defined(BSDI) || (defined(NETBSD) && __NetBSD_Version__ < 500000000) || defined(OPENBSD)
+#if defined(LINUX) || defined(AIX) || defined(BEOS) || defined(BSDI) || (defined(NETBSD) && __NetBSD_Version__ < 500000000) || defined(OPENBSD) || defined(__GLIBC__)
 
 #include <unistd.h>
 #include <fcntl.h>
Index: nss/nss/lib/ssl/sslmutex.h
===================================================================
--- nss.orig/nss/lib/ssl/sslmutex.h	2014-01-13 07:31:45.939967858 +0900
+++ nss/nss/lib/ssl/sslmutex.h	2014-01-13 07:31:45.935967956 +0900
@@ -50,7 +50,7 @@
 
 typedef int    sslPID;
 
-#elif defined(LINUX) || defined(AIX) || defined(BEOS) || defined(BSDI) || (defined(NETBSD) && __NetBSD_Version__ < 500000000) || defined(OPENBSD)
+#elif defined(LINUX) || defined(AIX) || defined(BEOS) || defined(BSDI) || (defined(NETBSD) && __NetBSD_Version__ < 500000000) || defined(OPENBSD) || defined(__GLIBC__)
 
 #include <sys/types.h>
 #include "prtypes.h"
Index: nss/nss/coreconf/arch.mk
===================================================================
--- nss.orig/nss/coreconf/arch.mk	2014-01-13 07:31:45.939967858 +0900
+++ nss/nss/coreconf/arch.mk	2014-01-13 07:31:45.935967956 +0900
@@ -122,6 +122,14 @@
     ifneq ($(words $(OS_RELEASE)),1)
 	OS_RELEASE := $(word 1,$(OS_RELEASE)).$(word 2,$(OS_RELEASE))
     endif
+    KERNEL = linux
+endif
+
+# This check must be last.  Since all uses of OS_ARCH that follow affect only
+# userland, we can merge other Glibc systems with Linux here.
+ifneq (, $(filter GNU GNU_%, $(OS_ARCH)))
+OS_ARCH = Linux
+OS_RELEASE = 2.6
 endif
 
 #
Index: nss/nss/coreconf/Linux.mk
===================================================================
--- nss.orig/nss/coreconf/Linux.mk	2014-01-13 07:31:45.939967858 +0900
+++ nss/nss/coreconf/Linux.mk	2014-01-13 07:31:45.935967956 +0900
@@ -93,7 +93,7 @@
 LIBC_TAG		= _glibc
 endif
 
-ifeq ($(OS_RELEASE),2.0)
+ifeq ($(KERNEL)-$(OS_RELEASE),linux-2.0)
 	OS_REL_CFLAGS	+= -DLINUX2_0
 	MKSHLIB		= $(CC) -shared -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so) $(RPATH)
 	ifdef MAPFILE
@@ -128,14 +128,21 @@
 # -ansi on platforms like Android where the system headers are C99 and do
 # not build with -ansi.
 STANDARDS_CFLAGS	= -D_POSIX_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE
-OS_CFLAGS		= $(STANDARDS_CFLAGS) $(DSO_CFLAGS) $(OS_REL_CFLAGS) $(ARCHFLAG) -Wall -Werror-implicit-function-declaration -Wno-switch -pipe -DLINUX -Dlinux -DHAVE_STRERROR
+OS_CFLAGS		= $(STANDARDS_CFLAGS) $(DSO_CFLAGS) $(OS_REL_CFLAGS) $(ARCHFLAG) -Wall -Werror-implicit-function-declaration -Wno-switch -pipe -DHAVE_STRERROR
+ifeq ($(KERNEL),linux)
+OS_CFLAGS		+= -DLINUX -Dlinux
+endif
 OS_LIBS			= $(OS_PTHREAD) -ldl -lc
 
 ifdef USE_PTHREADS
 	DEFINES		+= -D_REENTRANT
 endif
 
-ARCH			= linux
+ifeq ($(KERNEL),linux)
+	ARCH		= linux
+else
+	ARCH		= gnu
+endif
 
 DSO_CFLAGS		= -fPIC
 DSO_LDOPTS		= -shared $(ARCHFLAG)
@@ -152,7 +159,7 @@
 LDFLAGS			+= -Wl,-rpath-link,/usr/lib:/lib
 endif
 
-# INCLUDES += -I/usr/include -Y/usr/include/linux
+# INCLUDES += -I/usr/include
 G++INCLUDES		= -I/usr/include/g++
 
 #
@@ -187,7 +194,9 @@
 endif
 endif
 
+ifeq ($(KERNEL), linux)
 OS_REL_CFLAGS   += -DLINUX2_1
+endif
 MKSHLIB         = $(CC) $(DSO_LDOPTS) -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so) $(RPATH)
 
 ifdef MAPFILE
Index: nss/nss/coreconf/config.mk
===================================================================
--- nss.orig/nss/coreconf/config.mk	2014-01-13 07:31:45.939967858 +0900
+++ nss/nss/coreconf/config.mk	2014-01-13 07:31:45.935967956 +0900
@@ -31,7 +31,7 @@
 #######################################################################
 
 TARGET_OSES = FreeBSD BSD_OS NetBSD OpenUNIX OS2 QNX Darwin BeOS OpenBSD \
-              AIX RISCOS WINNT WIN95 Linux Android
+              AIX RISCOS WINNT WIN95 Linux Android GNU GNU_%
 
 ifeq (,$(filter-out $(TARGET_OSES),$(OS_TARGET)))
 include $(CORE_DEPTH)/coreconf/$(OS_TARGET).mk
